plugins/inputs/suricata/suricata.go

```diff
@@ -215,6 +227,28 @@ func (s *Suricata) parse(acc telegraf.Accumulator, sjson []byte) {
 	}
 }
 
+func (s *Suricata) parse(acc telegraf.Accumulator, sjson []byte) {
+	// initial parsing
+	var result map[string]interface{}
+	err := json.Unmarshal(sjson, &result)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+	// check for presence of relevant stats or alert
+	_, ok := result["stats"]
+	_, ok2 := result["alert"]
+	if !ok && !ok2 {
+		s.Log.Debug("Input does not contain necessary 'stats' or 'alert' sub-object")
+		return
+	}
+	if ok {
+		s.parseStats(acc, result)
+	} else if ok2 && s.Alerts {
+		s.parseAlert(acc, result)
+	}
```

Comment:
I think you should only check for the presence of the `alerts` field if you want to parse them...
```suggestion
	if _, ok := result["stats"]; !ok {
		s.Log.Debugf("Invalid input without 'stats': %v", result)
		return fmt.Errorf("input does not contain 'stats' object")
	}
	if _, ok2 := result["alert"]; s.Alerts && !ok2 {
		s.Log.Debugf("Invalid input without 'alert' when trying to parse them: %v", result)
		return fmt.Errorf("input does not contain 'alert' object")
	}
	
	s.parseStats(acc, result)
	if s.Alerts {
		s.parseAlert(acc, result)
	}
```

Reply:
@srebhan With your solution we force that "stats" event type must be always present. I have seen in some light environments that only "alert" types are configured as outputs in suricata. What do you think on making result["stats"] mandatory?

Reply:
Indeed, the Suricata event_type can only be of one type.

Reply:
Understood, so you either see `stats` **OR** `alerts`. I this correct? If so, please mark this as resolved even though I don't like `ok2`... :-)

Reply:
Exactly. Neither do I. I we add more fields we will have to be sure to properly handle them. Thanks.